src/services/task-scheduler-service.ts

```diff
@@ -47,22 +54,37 @@ export class TaskSchedulerService {
     )
 
     this._scheduledTask$ = taskWithRetryOnError$.pipe(
-      expand(() => {
+      expand((success: boolean) => {
         if (this._controller.signal.aborted) {
           return EMPTY
         }
 
+        if (cbAfterFailure && !success) {
+          logger.imp(`Last run of the task was not successful. Stopping the task scheduler`)
+          return EMPTY
+        }
+
         return timer(intervalMS).pipe(concatMap(() => taskWithRetryOnError$))
       }),
       share()
     )
 
-    this._scheduledTask$.subscribe()
+
+    this._scheduledTask$.subscribe({
+      complete: async () => {
+        if (!this._completed) {
```

Comment:
Does it matter that `_completed` could get set to `true` even if no `cbAfterFailure` was specified?

Reply:
This was added because I found a bug when calling stop on the task scheduler. But this reminded me of the rxjs property `Subscription.closed` which makes this not needed.